frontend/src/NavigationMenu/NavigationMenu.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import { List, ListItem, ListItemButton, ListItemText, Box, Drawer } from '@mui/material'
+import { UserType, UserTypeProps } from '../types/global.d'
+import { useTranslation } from 'react-i18next'
+
+export default function NavigationMenu(props: UserTypeProps) {
```

Comment:
I am not completely sure about this, but I think, classes are better way for writing React components. Because currently it's way too much code to place it in one function. Also, IIRC, classes allow to create so-called "hooks", which is a good React feature. And functions cannot do this...
@nebolax could you give any comments on this?

Reply:
According to the styling guide that we are using:
![image](https://user-images.githubusercontent.com/63492346/215055097-7e22a7c3-586f-4fe2-9a44-fed23dc795f5.png)

There is no need to save any state in `NavigationMenu`, right? So I would use a functional component.

But generally it's just a question of styling. Hooks, state, etc. are all doable in functional components too.

Reply:
Update: hooks is a functional-components-only thing. It provides same functionality as classes' `state`.